src/jit/decomposelongs.cpp

```diff
@@ -969,6 +969,45 @@ GenTree* DecomposeLongs::DecomposeMul(LIR::Use& use)
 }
 
 //------------------------------------------------------------------------
+// DecomposeUMod: Decompose GT_UMOD. The only GT_UMODs that make it to decompose
+// are guaranteed to be an unsigned long mod with op2 which is a cast to long from
+// a constant int whose value is between 2 and 0x3fffffff. All other GT_UMODs are
+// morphed into helper calls. These GT_UMODs will actually return an int value in
+// RDX. In decompose, we change the type of the operation to a TYP_INT, and replace
+// op2 with its low half.
+//
+// Arguments:
+//    use - the LIR::Use object for the def that needs to be decomposed.
+//
+// Return Value:
+//    The next node to process.
+//
+GenTree* DecomposeLongs::DecomposeUMod(LIR::Use& use)
+{
+    assert(use.IsInitialized());
+
+    GenTree*   tree = use.Def();
+    genTreeOps oper = tree->OperGet();
+
+    assert(oper == GT_UMOD);
```

Comment:
Should I also assert that loOp1 is between 2 and 0x3fffffff? Morph excludes everything that isn't in that range, but it would probably keep us safer just in case.